src/python/python-project.ts

```diff
@@ -222,6 +225,15 @@ export class PythonProject extends GitHubProject {
       );
     }
 
+    let poetry = options.poetry ?? false;
+    let pip = options.pip ?? true;
+    let venv = options.venv ?? true;
+    let setuptools = options.setuptools ?? this.projectType === ProjectType.LIB;
+    // If we select poetry, then pip/venv/setuptools must be set to false
+    if (options.poetry) {
+      pip = venv = setuptools = false;
+    }
```

Comment:
I think I'd rather have this explicitly erroring if an invalid combination is provided.
```suggestion
    const poetry = options.poetry ?? false;
    const pip = options.pip ?? !poetry;
    const venv = options.venv ?? !poetry;
    const setuptools = options.setuptools ?? !poetry && this.projectType === ProjectType.LIB;
```